### PR TITLE
TILA-2132 | Email template filters

### PR DESCRIPTION
**Pull request by nicobav**
## Change log
- Change email template rendering to be done with jinj2's sandboxed environment
- Adds filter support to email templates
- Adds currency filter
    - Formats numbers wit two decimal places with comma as separator and thousands separator is space
- Rename EMAIL_TEMPLATE_CONTEXT_ATTRS to EMAIL_TEMPLATE_CONTEXT_VARIABLES

Refs TILA-2132

### CODE REVIEW

File: email_notification/sender/email_notification_builder.py [hunk 1]
```diff
@@ -210,31 +213,25 @@ def _init_context_attr_map(self):
...
-                context_dict[key] = value()
-                continue
-
-            context_dict[key] = getattr(self.reservation, value, None)
-
-        return Context(context_dict)
+        validator.validate_string(
+            self.template.subject, self.context_attr_map, env=self.env
+        )
+        validator.validate_string(
```
Comment (saaste) [inline email_notification/sender/email_notification_builder.py:247]: `validator.validate_string` is called twice here.

File: email_notification/sender/email_notification_builder.py [hunk 2]
```diff
@@ -11,28 +11,44 @@
...
+        env.filters[fil] = func
+
     return env
 
 
 class EmailTemplateValidator:
     @property
     def bracket_lookup(self):
-        return re.compile(r"{{ *(\w+) *}}")
+        return re.compile(r"{{ *(\w+) \| *(\w+) *}}|{{ *(\w+) *}}")
```
Comment (saaste) [inline email_notification/sender/email_notification_builder.py:33]: Have to admit I have hard time understand what this regex should match 😄

File: email_notification/templatetags/email_template_filters.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from django import template
+
+from utils.decimal_utils import round_decimal
+
+register = template.Library()
+
+
+def format_currency(price: Decimal):
+    price = round_decimal(Decimal(price), 2)
+    kettu = "{:,.2f}".format(price).replace(",", " ").replace(".", ",")
```
Comment (saaste) [inline email_notification/templatetags/email_template_filters.py:12]: 🦊

**Review by saaste — CHANGES_REQUESTED** [review]

File: email_notification/templatetags/email_template_filters.py (hunk 3 above)
Comment (nicobav) [reply email_notification/templatetags/email_template_filters.py:12]: oh yes the fox! I'll clean this

File: email_notification/sender/email_notification_builder.py (hunk 1 above)
Comment (nicobav) [reply email_notification/sender/email_notification_builder.py:247]: good catch, poorly rebased, I'll clean this

File: email_notification/sender/email_notification_builder.py (hunk 2 above)
Comment (nicobav) [reply email_notification/sender/email_notification_builder.py:33]: words within {{}} and {{ | }}

File: email_notification/sender/email_notification_builder.py (hunk 1 above)
Comment (nicobav) [reply email_notification/sender/email_notification_builder.py:247]: oh, I backup a bit, there is a validation check for subject and content (the html content is its own content) so it is supposed be like this.

File: email_notification/templatetags/email_template_filters.py (hunk 3 above)
Comment (nicobav) [reply email_notification/templatetags/email_template_filters.py]: cleaned 🧹

File: email_notification/sender/email_notification_builder.py (hunk 1 above)
Comment (saaste) [reply email_notification/sender/email_notification_builder.py:247]: Funny, I was thinking that these are NOT the same line. To be sure to not look like a blind idiot, I read them like 100 times. I was sure they are identical.

I just learned I'm a blind idiot 🧑‍🦯🤪

**Review by saaste — APPROVED** [review]

**Pull request merged**